api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+
+/*The context class provides a context identifier*/
+class Context{
+  private:
+    /*The identifier itself*/
+    std::map<std::string, int> ctx_;
+
+  public:
+    /* Context: contructor, creates a context object from a map
+     * of keys and identifiers
+     */
+    Context(std::map<std::string, int> ctx);  
```

Comment:
All the public APIs need to be ABI compatible - which means they are not supposed to expose dependency on STL traits/layout.